components/flex/nz-flex.directive.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/LICENSE
+ */
+
+import { Directive, Input } from '@angular/core';
+
+import { nzAlign, nzDirection, nzFlex, nzGap, nzJustify, nzWrap } from './typings';
+
+@Directive({
+  selector: '[nz-flex],nz-flex',
+  exportAs: 'nzFlex',
+  host: {
+    class: 'ant-flex',
+    '[style.--flex-direction]': `nzDirection`,
+    '[style.--flex-justify]': `nzJustify`,
+    '[style.--flex-align]': `nzAlign`,
+    '[style.--flex-gap]': `nzGap === 'small' ? '8px' : nzGap === 'middle'? '16px': nzGap === 'large' ? '24px' : nzGap + 'px'`,
+    '[style.--flex-wrap]': `nzWrap`,
+    '[style.--flex]': `nzFlex`
+  }
+})
+export class NzFlexDirective {
```

Comment:
Can we make it a standalone component? Reference: #8037

Reply:
I will check it out and get back to you as soon as possible.

Reply:
I have made it a standalone directive. Please check it out and let me know if you have any feedback.